sonic-sensormond/scripts/sensormond

```diff
@@ -0,0 +1,559 @@
+
+    return ret
+
+def update_entity_info(table, parent_name, key, device, device_index):
+    fvs = swsscommon.FieldValuePairs(
+        [('position_in_parent', str(try_get(device.get_position_in_parent, device_index))),
+         ('parent_name', parent_name)])
+    table.set(key, fvs)
+
+class SensorStatus(logger.Logger):
+
+    def __init__(self):
+        super(SensorStatus, self).__init__(SYSLOG_IDENTIFIER)
+
+        self.value = None
+        self.over_threshold = False
+        self.under_threshold = False
+
+    def set_value(self, name, value):
+        """
+        Record sensor changes.
+        :param name: Name of the sensor.
+        :param value: New value.
+        """
+        if value == NOT_AVAILABLE:
+            if self.value is not None:
+                self.log_warning('Value of {} became unavailable'.format(name))
+                self.value = None
+            return
+
+        self.value = value
+
+    def set_over_threshold(self, value, threshold):
+        """
+        Set over threshold status
+        :param value: value
+        :param threshold: High threshold
+        :return: True if over threshold status changed else False
+        """
+        if value == NOT_AVAILABLE or threshold == NOT_AVAILABLE:
+            self.log_warning('Value/threshold of {} became unavailable {}/{}'.format(self.name, value, threshold))
+            old_status = self.over_threshold
+            self.over_threshold = False
+            return old_status != self.over_threshold
+
+        status = value > threshold
+        if status == self.over_threshold:
+            return False
+
+        self.over_threshold = status
+        return True
+
+    def set_under_threshold(self, value, threshold):
+        """
+        Set under value status
+        :param value: value
+        :param threshold: Low threshold
+        :return: True if under threshold status changed else False
+        """
+        if value == NOT_AVAILABLE or threshold == NOT_AVAILABLE:
+            old_status = self.under_threshold
+            self.under_threshold = False
+            return old_status != self.under_threshold
+
+        status = value < threshold
+        if status == self.under_threshold:
+            return False
+
+        self.under_threshold = status
+        return True
+
+
+#
+# voltageUpdater  ======================================================================
+#
+class VoltageUpdater(logger.Logger):
+    # Voltage information table name in database
+    VOLTAGE_INFO_TABLE_NAME = 'VOLTAGE_INFO'
+
+    def __init__(self, chassis):
+        """
+        Initializer of VoltageUpdater
+        :param chassis: Object representing a platform chassis
+        """
+        super(VoltageUpdater, self).__init__(SYSLOG_IDENTIFIER)
+
+        self.chassis = chassis
+        self.voltage_status_dict = {}
+        state_db = daemon_base.db_connect("STATE_DB")
+        self.table = swsscommon.Table(state_db, VoltageUpdater.VOLTAGE_INFO_TABLE_NAME)
```

Comment:
Might be better to declare table constants in the beginning so you don't have to reference it with class name each time

Reply:
Class name isn't a requirement, can also be referenced using `self`

Reply:
ok.

Reply:
Not sure of the change you want to make here (if any), but either the class name or `self` are ok. I wouldn't declare these are file-scope constants just to remove the class/self, since its scope should be limited to this class.

Reply:
I used self :). its updated in the diffs.